### DISCUSSION TITLE
Add possibility to provide new functions like `utf8` and `ascii`.

### DISCUSSION THREAD

**Discussion by mainrs** [hlorenzi/customasm#156] [Ideas]
Some systems rely on obscure string encodings. A mechanism to provide custom formatters would be a nice feature to have. Maybe it could be based on WASM? And customasm provides a library that such formatters compile against.

One could the tell customasm to load these formatters: `customasm --formatters <file>,<dir> ...`. Maybe this could be turned into customasm functioning more as an "engine" of some kind.

**Comment by hlorenzi** [comment]
I had the intention of adding support for a `charmap` directive, which I've seen other assemblers do. Then you could specify a translation table for each character (or range of characters) into your specific encoding -- much like we already do with instructions in a `ruledef` block. 

Something like [this](https://github.com/pret/pokered/blob/master/charmap.asm). Do you think that would be enough?

**Reply by mainrs** [reply]
Pokemon was the example I had in mind :) I think that would be enough for this use-case, definitely!
